services/storage/src/simcore_service_storage/utils.py

```diff
@@ -55,4 +57,15 @@ def expo(base=1.2, factor=0.1, max_value=2):
             yield a
             n += 1
         else:
-            yield max_value
\ No newline at end of file
+            yield max_value
+
+
+async def download_to_file_or_raise(
+    session: ClientSession, url: StrOrURL, destination_path: str
+):
+    # FIXME: stream chunks
+
+    # raises aiohttp.ClientResponseError
+    async with session.get(url, raise_for_status=True) as resp:
+        async with aiofiles.open(destination_path, mode="wb") as fh:
+            await fh.write(await resp.read())
```

Comment:
please use a chunked writer to avoid memory issues. Also make sure this client session has a timeout, maybe one of those 1 hour timeouts we use across the platform which is tied to an environment variable

Reply:
Yes, this is what I meant in the FIXME note.
Take into account that storage is a very old code. I had to put this part of the code aside so it could be testable ( I needed to mock downloading from DATCore service).
I did not want to add more changes **in this PR** than the strictly related to the subject of the PR ... so we do not loose focus on the fix we are working here.

In any case, since this is isolated and you insist, I will add it now.